File: ssr-simple/src/main/java/com/cincinnatiai/ssr_simple/model/TableColumnModel.kt
package com.cincinnatiai.ssr_simple.model

data class TableColumnModel(
    val header: String,
    val weight: Float? = null, // Column weight for flexible width (e.g., 1f, 2f)
    val width: Int? = null, // Fixed width in dp (overrides weight if set)
    val horizontalAlignment: String? = null, // "start", "center", "end"
    val textStyle: TextStyleModel? = null, // Style for this column's cells
    val headerStyle: TextStyleModel? = null // Style for this column's header
)

File: ssr-simple/ssr-simple-react/src/models/TableColumnModel.ts
import { TextStyleModel } from './TextStyleModel';

export interface TableColumnModel {
  header: string;
  weight?: number; // Column weight for flexible width (e.g., 1, 2)
  width?: number; // Fixed width in px (overrides weight if set)
  horizontalAlignment?: 'start' | 'center' | 'end' | 'left' | 'right';
  textStyle?: TextStyleModel; // Style for this column's cells
  headerStyle?: TextStyleModel; // Style for this column's header
}
